### ISSUE TITLE
Problem for using only step 6 to get the outputs

### ISSUE THREAD

**Issue by MianMianMeow** [Tai-Hsien/MeshSegNet#27]
Hello  Tai,
  Really appreciate for your research~ I'm a beginner of mesh segmentation and trying to use  **step6_predict_with_post_processing_pygco.py**  and your pre-trained model to get an output. I haven't tried to use my own data to train the model yet.

1. The input STL is **Example_02.zip** shared by @smoothumut from #6 , and there's no error reported while running **step6_predict_with_post_processing_pygco.py**. 
2. I opened the output .vtp file in **ParaView** and it looks below:

![outputs](https://user-images.githubusercontent.com/86562720/142583326-c2b939c2-d555-4cb1-8729-4e1ae859f00c.jpg)

3. I noticed the pre-trained models are with suffix .zip, but in **step6_predict_with_post_processing_pygco.py** it has _model_name = 'MeshSegNet_Max_15_classes_72samples_lr1e-2_best.tar'_ . I renamed the model from .zip to .tar but it still does not work.

Do you have any hints for my wrong outputs?

**Comment by MianMianMeow** [comment]
Hello Tai,
I noticed the problem may caused by the conversion of input file from .stl to .vtp. 
1. For the output shown above, I used the this script to do the conversion:  
`import time
from io import open

import sys
import vtk
import numpy as np
from stl import mesh

def stl2vtk(data):
    surface = vtk.vtkPolyData()
    points = vtk.vtkPoints()
    polys = vtk.vtkCellArray()

    s = data.vectors.shape
    n = s[0]
    X = data.vectors.reshape([s[0] * s[1], s[2]])
    for i, p in enumerate(X):
        points.InsertPoint(i, p)

    for i in range(n):
        polys.InsertNextCell(3, [3*i, 3*i+1, 3*i+2])
    
    surface.SetPoints(points)
    surface.SetPolys(polys)
    
    return surface

def save_vtk(object, filename):
    writer = vtk.vtkXMLPolyDataWriter()
    writer.SetFileName(filename)
    writer.SetInputData(object)
    writer.Write()


if __name__ == '__main__':
    
    t0 = time.time()
    
    # if(len(sys.argv) != 3):
    #     print("Usage: %s infile.stl outfile.vtp\n" % sys.argv[0])
    #     exit(-1)
        
    data = mesh.Mesh.from_file("D:\\project\\MeshSegNet\\issues\\Example_02.stl")
    surface = stl2vtk(data)
    save_vtk(surface, "D:\\project\\MeshSegNet\\issues\\Example_02.vtp")
    
    t1 = time.time()
    print("Time for data conversion: {0:.3f} seconds.\n".format(t1-t0))`

2.  For the output shown below, I used ParaView to open the .stl file and then save it as .vtp file. This output seems better although it still does not have the labels for each tooth. Not sure if I should do some configurations in ParaView to show the correct outputs with labels.
![ParaView_output](https://user-images.githubusercontent.com/86562720/142587018-5dcc06c7-3ee3-42e3-ac28-c94915616b40.png)
.

Could you please share your method to convert the input file from .stl to .vtp ?

**Comment by Tai-Hsien** [comment]
Hello,

Sorry for the late reply.

I attached a result that from the step6 with the "Example_02.stl" as you mentioned.
![image](https://user-images.githubusercontent.com/9615481/142717228-65db2eb6-e3b7-4702-b67f-795a384dd352.png)

Yes, we need to modify the extension file format of pretrained model from tar to zip. The figure you showed above was draw by "Normal" array instead of "Label". You can change it easy in Paraview.

Also, step 6 load a STL file and generate a VTP output, so you don't need to convert STL to VTP if you want to use step 6.

Hope it helps.

Best,

**Comment by MianMianMeow** [comment]
> The figure you showed above was draw by "Normal" array instead of "Label". You can change it easy in Paraview.

Wow~ This exactly solves my Problem. Thanks a lot.
